charts/apisix/templates/deployment.yaml

```diff
@@ -65,9 +65,11 @@ spec:
             - name: http
               containerPort: {{ .Values.gateway.http.containerPort }}
               protocol: TCP
+              hostPort: {{ .Values.gateway.http.hostPort }} 
```

Comment:
Here we should add an `if` block to check whether the `hostPort` is configured. Or hostPort is always enabled.